android/src/main/java/app/notifee/core/NotifeeAlarmManager.java

```diff
@@ -159,12 +159,45 @@ static void scheduleTimestampTriggerNotification(
     // Ensure timestamp is always in the future when scheduling the alarm
     timestampTrigger.setNextTimestamp();
 
-    if (timestampTrigger.getAllowWhileIdle()) {
-      AlarmManagerCompat.setExactAndAllowWhileIdle(
+    TimestampTriggerModel.AlarmType alarmType = timestampTrigger.getAlarmType();
+
+    switch (alarmType) {
+      case SET:
+        alarmManager.set(AlarmManager.RTC, timestampTrigger.getTimestamp(), pendingIntent);
+        break;
+      case SET_AND_ALLOW_WHILE_IDLE:
+        AlarmManagerCompat.setAndAllowWhileIdle(
           alarmManager, AlarmManager.RTC_WAKEUP, timestampTrigger.getTimestamp(), pendingIntent);
-    } else {
-      AlarmManagerCompat.setExact(
+        break;
+      case SET_EXACT:
+        AlarmManagerCompat.setExact(
+          alarmManager, AlarmManager.RTC_WAKEUP, timestampTrigger.getTimestamp(), pendingIntent);
+        break;
+      case SET_EXACT_AND_ALLOW_WHILE_IDLE:
+        AlarmManagerCompat.setExactAndAllowWhileIdle(
           alarmManager, AlarmManager.RTC_WAKEUP, timestampTrigger.getTimestamp(), pendingIntent);
+        break;
+      case SET_ALARM_CLOCK:
+        // probably a good default behavior for setAlarmClock's
+
+        int mutabilityFlag = PendingIntent.FLAG_UPDATE_CURRENT;
+        if (android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.M) {
+          mutabilityFlag = PendingIntent.FLAG_IMMUTABLE | PendingIntent.FLAG_UPDATE_CURRENT;
+        }
+
+        Context context = getApplicationContext();
+        Intent launchActivityIntent =
+          context.getPackageManager().getLaunchIntentForPackage(context.getPackageName());
+
+        PendingIntent pendingLaunchIntent =
+          PendingIntent.getActivity(
+            context,
+            notificationModel.getId().hashCode(),
+            launchActivityIntent,
+            mutabilityFlag);
+        AlarmManagerCompat.setAlarmClock(
```

Comment:
from what I remember, when using the `setAlarmClock` api, it may show an alarm icon is some versions and devices, that when you touch the icon, it will use the pending intent. and so I remember it being mandatory. I'm not sure if it is, but as far as I remember, it was required.

Reply:
so Even if it is not required, I'd like the icon to launch the app that has caused it to appear (as it is common UX for alarm apps I believe).